prow/cmd/deck/static/api/pr.ts

```diff
@@ -5,9 +5,29 @@ export interface Label {
   Name: string;
 }
 
+export interface Ref {
+  Name: string;
+  Prefix: string;
+}
+
+export interface Author {
+  Login: string;
+}
+
+export interface Repository {
+  Name: string;
+  NameWithOwner: string;
+  Owner: {
+    Login: string;
+  };
+}
+
 export interface PullRequest extends BasePullRequest {
   Merged: boolean;
   Title: string;
+  Author: Author;
```

Comment:
Note that there are two different `PullRequest` definition in prow, one for tide https://github.com/kubernetes/test-infra/blob/67b179342c2a6bd7a447b0e70cb94cf6d7fe0f55/prow/tide/tide.go#L1909 and one for the rest:https://github.com/kubernetes/test-infra/blob/67b179342c2a6bd7a447b0e70cb94cf6d7fe0f55/prow/prstatus/prstatus.go#L87

The same happens in Typescript with inheritance as you have noted here. The fields that are moved out of BasePullRequest is that they are not really used by tide at all, and doesn't make sense to be kept in Base struct, typescript compilation would catch error if this field is later on used. 

This might be a tech debt, but might not be an easy one to solve. I'd prefer not to fix this "tech debt" right now due to the time pressure of what I need to deliver